buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesInfoTask.java

```diff
@@ -0,0 +1,315 @@
+                continue;
+            }
+            // only external dependencies are checked
+            if (dependency.getGroup() != null && dependency.getGroup().contains("org.elasticsearch")) {
+                continue;
+            }
+
+            final String url = createURL(dependency.getGroup(), dependency.getName(), dependency.getVersion());
+            final String dependencyName = DependencyLicensesTask.getDependencyName(mappings, dependency.getName());
+            logger.info("mapped dependency " + dependency.getGroup() + ":" + dependency.getName() + " to " + dependencyName + " for license info");
+
+            final String licenseType = getLicenseType(dependency.group, dependencyName);
+            output.append(dependency.getGroup() + ":" + dependency.getName() + "," + dependency.getVersion() + "," + url + "," + licenseType + "\n");
+
+        }
+        Files.write(outputfile.toPath(), output.toString().getBytes(StandardCharset.UTF_8));
+    }
+    
+    /**
+     * Create an URL on <a href="https://repo1.maven.org/maven2/">Maven Central</a>
+     * based on dependency coordinates.
+     */
+    protected String createURL(final String group, final String name, final String version){
+        final String baseURL = "https://repo1.maven.org/maven2";
+        return baseURL + "/" + group.replaceAll("\\.", "/") + "/" + name + "/" + version;
+    }
+    
+    /**
+     * Read the LICENSE file associated with the dependency and determine a license type.
+     *
+     * The license type is one of the following values:
+     * <u>
+     *     <li><em>UNKNOWN</em> if LICENSE file is not present for this dependency.</li>
+     *     <li><em>one SPDX identifier</em> if the LICENSE content matches with an SPDX license.</li>
+     *     <li><em>Custom;URL</em> if it's not an SPDX license,
+     *          URL is the Github URL to the LICENSE file in elasticsearch repository.</li>
+     * </ul>
+     *
+     * @param group dependency group
+     * @param name dependency name
+     * @return SPDX identifier, UNKNOWN or a Custom license
+     */
+    protected String getLicenseType(final String group, final String name) throws IOException {
+        File license;
+
+        if (licensesDir.exists()) {
+            File[] matchedfiles = licensesDir.listFiles((dir, name) -> name.contains("-LICENSE"));
+            for(File file : matchedfiles) {
+                String prefix = file.getName().split("-LICENSE.*")[0];
+                if (group.contains(prefix) || name.contains(prefix)) {
+                    license = file.getAbsoluteFile();
+                }
+            }
+        }
+
+        if (license != null) {
+            // replace * because they are sometimes used at the beginning lines as if the license was a multi-line comment
+            final String content = new String(Files.readAllBytes(outfile.toPath()), StandardCharsets.UTF_8);
+            content.replaceAll("\\s+", " ").replaceAll("\\*", " ");
```

Comment:
It kinda feels weird when invoking `replaceAll` on a `final` instance. Also `replaceAll` does not change the value of `content` and it returns the `updated` string and you are not storing it anywhere (hence changes lost). You might have to stick with existing solution 

```final String content = new String(license.readBytes(), "UTF-8").replaceAll("\\s+", " ").replaceAll("\\*", " ")``` 

It does not change the value of a `final` field as well as the result of `replaceAll` on string is not `lost` 👍

Reply:
Ups! My bad. Thanks for pointing that out. But isn't ```readBytes()``` [groovy](http://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/File.html#readBytes()) ? I replaced it with ```Files.readAllBytes()```